feeds/backend.php

```diff
@@ -30,21 +33,45 @@
         switch($content) {
             case "posted":
                 $condition = sprintf("state='%s'", PROJ_SUBMIT_PG_POSTED);
+                $desc = sprintf(_("The latest releases posted to Project Gutenberg from %1\$s."), $site_name);
                 break;
             case "postprocessing":
                 $condition = sprintf("state='%s'", PROJ_POST_FIRST_AVAILABLE);
+                $desc = sprintf(_("The latest releases available at %1\$s for post-processing."), $site_name);
                 break;
             case "proofing":
                 $condition = sprintf("state='%s'", PROJ_P1_AVAILABLE);
+                $desc = sprintf(_("The latest releases available at %1\$s for proofreading."), $site_name);
                 break;
             case "smoothreading":
-                $condition = "
-                    state = 'proj_post_first_checked_out' AND
-                    smoothread_deadline > UNIX_TIMESTAMP()";
+                $query = "
+                    SELECT 
+                        projectid, nameofwork, genre, language, FROM_UNIXTIME(e.timestamp)
+                    FROM projects
+                    JOIN project_events e USING (projectid)
+                    WHERE
+                        e.event_type = 'smooth-reading' AND
+                        e.details1 = 'text available' AND
+                        e.timestamp > UNIX_TIMESTAMP() - (30*24*60*60)
+                    ORDER BY e.timestamp DESC
+                    LIMIT $limit";
+                $desc = sprintf(_("The latest releases available at %1\$s for Smooth Reading."), $site_name);
                 break;
         }
+
+        // If $query is not set, use the default query.
+        if (!isset($query))
+        {
+            $query = "
+                SELECT * 
+                FROM projects 
+                WHERE $condition 
+                ORDER BY modifieddate DESC 
+                LIMIT $limit";
+        }
+
         $data = '';
-        $result = mysqli_query(DPDatabase::get_connection(), "SELECT * FROM projects WHERE $condition ORDER BY modifieddate DESC LIMIT 10");
+        $result = mysqli_query(DPDatabase::get_connection(), $query);
         while ($row = mysqli_fetch_array($result)) {
             $posteddate = date("r",($row['modifieddate']));
```

Comment:
`modifieddate` isn't being returned in the new SR query so this should be generating a NOTICE if `$testing = 1`. For SR, this should probably be the e.timestamp value. You could do this via:
```
SELECT 
    projectid, nameofwork, genre, language, FROM_UNIXTIME(e.timestamp) as modifieddate
```

Reply:
I modified the query to add `modifieddate` in 9ec53768b5bc3af587162cebb76e68cc2d1b508c, but I'm not sure what the "generating a NOTICE" part refers to.

Reply:
If you assign 'modifieddate' as an alias for FROM_UNIXTIME(e.timestamp), doesn't that have the potential of being confusing down the line, since modifieddate is an actual field name in another table, and the two are not necessarily the same? (Or am I being too picky?)

Reply:
I think you're being too picky here. :)